framework/src/task_manager/task_manager_core.c

```diff
@@ -732,6 +784,7 @@ static int taskmgr_getinfo_with_handle(int handle, tm_response_t *response_msg)
 
 static void taskmgr_termination_callback(void)
 {
+	task_manager_pid = TM_INVALID_PID;
```

Comment:
TM_INVALID_PID is not a good name for this case. TM_NOT_AVAILABLE or TM_DEAD or TM_NOT_ALIVE or something else.